### PR TITLE
[refactor] Rename the confused needs_grad

### PR DESCRIPTION
**Pull request by erizmr**
Related issue = #

### CODE REVIEW

**Review by ailzhang — COMMENTED** [review]
how about we just use `is_real`? :P

File: taichi/python/export_lang.cpp [hunk 1]
```diff
@@ -970,7 +970,7 @@ void export_lang(py::module &m) {
   });
 
   m.def("test_throw", [] { throw IRModified(); });
-  m.def("needs_grad", needs_grad);
+  m.def("is_real", is_real);
```
Comment (strongoier) [inline taichi/python/export_lang.cpp:973]: IIRC `is_real` is already exported :-)

Comment (erizmr) [reply taichi/python/export_lang.cpp:973]: Let me remove the redundant one.

**Review by strongoier — APPROVED** [review]
LGTM!

**Pull request merged**